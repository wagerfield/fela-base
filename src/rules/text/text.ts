import { Unit } from "../../types"

export interface TextRuleOptions {
  family?: string
  size?: string
  style?: string
  weight?: string
  color?: string
  align?: string
  margin?: Unit
  lineHeight?: Unit
  transform?: string
  ellipsis?: boolean
}

export default ({
  family,
  size,
  style,
  weight,
  color,
  align,
  margin,
  lineHeight,
  transform,
  ellipsis
}: TextRuleOptions = {}) => ({
  fontFamily: family,
  fontSize: size,
  fontStyle: style,
  fontWeight: weight,
  textAlign: align,
  textTransform: transform,
  lineHeight,
  ellipsis,
  margin,
  color
})
