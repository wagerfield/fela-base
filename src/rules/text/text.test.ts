import text from "./text"

describe("text(props)", () => {
  it("renders expected snapshots", () => {
    expect(text()).toMatchSnapshot()
    expect(
      text({
        family: "Helvetica",
        size: "24px",
        weight: "bold"
      })
    ).toMatchSnapshot()
  })
})
