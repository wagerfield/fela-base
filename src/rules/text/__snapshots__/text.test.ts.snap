// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text(props) renders expected snapshots 1`] = `
Object {
  "color": undefined,
  "ellipsis": undefined,
  "fontFamily": undefined,
  "fontSize": undefined,
  "fontStyle": undefined,
  "fontWeight": undefined,
  "lineHeight": undefined,
  "margin": undefined,
  "textAlign": undefined,
  "textTransform": undefined,
}
`;

exports[`text(props) renders expected snapshots 2`] = `
Object {
  "color": undefined,
  "ellipsis": undefined,
  "fontFamily": "Helvetica",
  "fontSize": "24px",
  "fontStyle": undefined,
  "fontWeight": "bold",
  "lineHeight": undefined,
  "margin": undefined,
  "textAlign": undefined,
  "textTransform": undefined,
}
`;
